SpatialGDK/Source/SpatialGDK/Private/Interop/SnapshotManager.cpp

```diff
@@ -143,7 +143,8 @@ void USnapshotManager::LoadSnapshot(const FString& SnapshotName)
 				// Entity component data must be deep copied so that it can be used for CreateEntityRequest.
 				Schema_ComponentData* CopySchemaData = DeepCopyComponentData(EntityToSpawn->components[i].schema_type);
 				Worker_ComponentData EntityComponentData{};
-				EntityComponentData.component_id = Schema_GetComponentDataComponentId(CopySchemaData);
+				//TODO: Check if this works
+				EntityComponentData.component_id = EntityToSpawn->components[i].component_id;
```

Comment:
Can't see why this wouldn't work although you may as well store a reference to the worker component object.